# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/PeerConnectionImpl.cpp
# Commit: ceed49af0fad
# Full Hash: ceed49af0fad6ee87f64a89adb84c791a74ced14
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Don't register/unregister conduit with the call wrapper in ctor/dtor. r=ng,bwc
#   
#   Doing this outside of the dtor, especially, removes the requirement of the
#   conduit having to be destroyed on the worker thread (main).
#   
# ==============================================================================

diff -r a1c16b86d3a7 -r ceed49af0fad dom/media/webrtc/jsapi/PeerConnectionImpl.cpp
--- a/dom/media/webrtc/jsapi/PeerConnectionImpl.cpp	Mon Jan 18 16:41:26 2021 +0100
+++ b/dom/media/webrtc/jsapi/PeerConnectionImpl.cpp	Mon Jan 18 17:32:49 2021 +0100
@@ -2573,7 +2573,7 @@
     auto report = MakeUnique<dom::RTCStatsCollection>();
     if (dom::RTCBandwidthEstimationInternal* bw =
             report->mBandwidthEstimations.AppendElement(fallible)) {
-      const auto& stats = aPipeline->Conduit()->GetCallStats();
+      const auto& stats = aPipeline->mConduit->GetCallStats();
       bw->mTrackIdentifier = trackName;
       bw->mSendBandwidthBps.Construct(stats.send_bandwidth_bps / 8);
       bw->mMaxPaddingBps.Construct(stats.max_padding_bitrate_bps / 8);
@@ -2590,9 +2590,9 @@
   using TimeStampPromise = MozPromise<Maybe<DOMHighResTimeStamp>, bool, true>;
   promises.AppendElement(
       InvokeAsync(mSTSThread, __func__,
-                  [cond = RefPtr<MediaSessionConduit>(aPipeline->Conduit())] {
+                  [conduit = aPipeline->mConduit] {
                     return TimeStampPromise::CreateAndResolve(
-                        cond->LastRtcpReceived(), __func__);
+                        conduit->LastRtcpReceived(), __func__);
                   })
           ->Then(
               GetMainThreadSerialEventTarget(), __func__,
@@ -2602,8 +2602,8 @@
                     aValue.ResolveValue();
 
                 auto report = MakeUnique<dom::RTCStatsCollection>();
-                auto asAudio = aPipeline->Conduit()->AsAudioSessionConduit();
-                auto asVideo = aPipeline->Conduit()->AsVideoSessionConduit();
+                auto asAudio = aPipeline->mConduit->AsAudioSessionConduit();
+                auto asVideo = aPipeline->mConduit->AsVideoSessionConduit();
 
                 nsString kind = asVideo.isNothing() ? u"audio"_ns : u"video"_ns;
                 nsString idstr = kind + u"_"_ns;
@@ -2615,7 +2615,7 @@
                 nsString remoteId;
                 Maybe<uint32_t> ssrc;
                 std::vector<unsigned int> ssrcvals =
-                    aPipeline->Conduit()->GetLocalSSRCs();
+                    aPipeline->mConduit->GetLocalSSRCs();
                 if (!ssrcvals.empty()) {
                   ssrc = Some(ssrcvals[0]);
                 }